File: CSharp/Parrot/NorwegianParrot.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Parrot
{
    public class NorwegianParrot
    {
        private bool IsNailed;
        private double Voltage;

        public NorwegianParrot(double voltage, bool isNailed)
        {
            IsNailed = isNailed;
            Voltage = voltage;
        }

        public double GetSpeed()
        {
            return IsNailed ? 0 : Math.Min(24.0, Voltage * NewParrot.BaseSpeed);
        }

        public string GetCry()
        {
            return Voltage > 0 ? "Bzzzzzz" : "...";
        }
    }
}

File: Java/src/main/com/parrot/NorwegianParrot.java
package com.parrot;
public class NorwegianParrot extends Parrot {
    public NorwegianParrot(int numberOfCoconuts, double voltage, boolean isNailed) {
        super(numberOfCoconuts, voltage, isNailed);
    }

    @Override
    public double getSpeed() {
        return  (isNailed) ? 0 : Math.min(MAX_SPEED, voltage * BASE_SPEED );
    }

    @Override
    protected String getCry() {
        return (voltage > 0) ? "Bzzzzzz" : "...";
    }
}
